File: baekjoon/11809.java
import java.util.*;

class Main{

    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        String A = scan.next();
        String B = scan.next();
        solve(A, B);
    }

    private static void solve(String A, String B)
    {
        int a_idx = A.length() - 1;
        int b_idx = B.length() - 1;
        String AA = A;
        String BB = B;

        while(a_idx >= 0 && b_idx >= 0)
        {
            if(A.charAt(a_idx) < B.charAt(b_idx))
            {
                AA = erase(AA, a_idx);
            }
            else if(A.charAt(a_idx) > B.charAt(b_idx))
            {
                BB = erase(BB, b_idx);
            }

            --a_idx;
            --b_idx;
        }

        print_to_number(AA);
        print_to_number(BB);
    }

    private static String erase(String S, int idx)
    {
        String A = S.substring(0, idx);
        String B = S.substring(idx + 1, S.length());
        return A + B;
    }

    private static void print_to_number(String S)
    {
        int num = 0, idx = S.length() - 1;
        int radix = 1;

        while(idx >= 0)
        {
            num += (S.charAt(idx) - '0') * radix;
            radix *= 10;
            --idx;
        }

        System.out.println(S.length() == 0 ? "YODA" : num);
    }
}

File: baekjoon/11809.cpp
#include <iostream>
#include <string>
using namespace std;

void print_to_number(const string& S)
{
    if(S.empty())
    {
        cout << "YODA\n";
        return;
    }

    int num = 0, idx = S.size() - 1;
    int radix = 1;

    while(idx >= 0)
    {
        num += (S[idx] - '0') * radix;
        radix *= 10;
        --idx;
    }

    cout << num << endl;
}

void solve(const string& A, const string& B)
{
    string AA(A), BB(B);
    int a_idx = A.size() - 1, b_idx = B.size() - 1;

    while(a_idx >= 0 && b_idx >= 0)
    {
        if(A[a_idx] < B[b_idx])
        {
            AA.erase(a_idx, 1);
        }
        else if(A[a_idx] > B[b_idx])
        {
            BB.erase(b_idx, 1);
        }

        --a_idx;
        --b_idx;
    }
    print_to_number(AA);
    print_to_number(BB);
}

void proc()
{
    string A, B;
    cin >> A >> B;
    solve(A, B);
}

int main()
{
    proc();
    return 0;
}
